runtime/util/hshelp.c

```diff
@@ -3229,6 +3230,20 @@ classIsModifiable(J9JavaVM * vm, J9Class * clazz)
 		rc = JNI_FALSE;
 	} else if ((J2SE_VERSION(vm) >= J2SE_19) && J9_ARE_ALL_BITS_SET(clazz->classFlags, J9ClassIsAnonymous)) {
 		rc = JNI_FALSE;
+	} else {
+		/* Handle name based matches */
+		U_8* className = J9UTF8_DATA(J9ROMCLASS_CLASSNAME(clazz->romClass));
+		U_8 classNameLength = J9UTF8_LENGTH(className);
```

Comment:
This should be `J9UTF8_LENGTH(J9ROMCLASS_CLASSNAME(clazz->romClass));`

The length isn't available from the data.

Reply:
updated